### DISCUSSION TITLE
Livewire Best Practices

### DISCUSSION THREAD

**Discussion by michael-rubel** [livewire/livewire#4912] [2. Show and tell]
Hi, @calebporzio, @joshhanley & The Livewire Folks!

Great community, great framework!
I'm not ashamed to say that this tool has made my career alongside Laravel itself, as I use it at work on a daily basis starting from early 2021, and for hobby projects starting from 2019. As a backend-oriented developer, in my experience, I haven't met any framework issues I was unable to resolve. Some internal issues were PRed by @joshhanley faster than I could, and I'm very grateful to you guys for your work, and I wish the framework grows even faster!

Meanwhile, some issues still persist, but most of them persist due to the nature of how Livewire works. To avoid problems you just need to consider some facts. I often hear people say like "livewire wasn't designed for enterprise", "performance is poor", or "security is questionable". It doesn’t have to be this way... That's why I created [Livewire Best Practices](https://github.com/michael-rubel/livewire-best-practices), an open-source repository that I can share to help people to understand the pitfalls of the Livewire framework, and how to deal with them.

I ask you guys to help me extend or correct the list of available tips, based on your own experience, and, if the community accepts, we would PR it to ReadMe alongside "awesome-livewire".

Kind regards,
Michael

**Comment by joshhanley** [comment]
@michael-rubel thanks for sharing! Yep, conceptually, Livewire can be tough at times.

I think it could also be useful to improve the docs, PR's are always welcome to update them https://github.com/livewire/docs

**Reply by michael-rubel** [reply]
@joshhanley Wow, I see @calebporzio now [working](https://github.com/livewire/livewire/commit/27e8de90dfd10dcd61e9e34e9ab486c35488a528) on support Form Requests internally after someone posted my tip on Twitter. 😮

Can I PR this list to the ReadMe so maybe it will be useful for someone?

**Reply by joshhanley** [reply]
@michael-rubel oh nice! Yep PR's are welcome, but it will be up to Caleb to decide whether he wants to merge it or not 🙂
